server/sonar-server/src/main/java/org/sonar/server/user/ws/UserJsonWriter.java

```diff
@@ -94,7 +93,7 @@ public void write(JsonWriter json, @Nullable UserDto user) {
   }
 
   private void writeGroupsIfNeeded(JsonWriter json, Collection<String> groups, @Nullable Collection<String> fields) {
-    if (isFieldNeeded(FIELD_GROUPS, fields) && userSession.hasPermission(GlobalPermissions.SYSTEM_ADMIN)) {
+    if (isFieldNeeded(FIELD_GROUPS, fields) && userSession.isRoot()) {
```

Comment:
This class is used by administration console of users, not by users to know their groups:
* api/users/deactivate
* api/users/search
* api/users/update

So it's the expected behavior.